src/SecretManager/Program.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Framework.ConfigurationModel.UserSecrets;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Runtime.Common.CommandLine;
+using Newtonsoft.Json.Linq;
+
+namespace SecretManager
+{
+    public class Program
+    {
+        private ILogger _logger;
+        private CommandOutputProvider _loggerProvider;
+
+        public Program()
+        {
+            var loggerFactory = new LoggerFactory();
+            CommandOutputProvider = new CommandOutputProvider();
+            loggerFactory.AddProvider(CommandOutputProvider);
+            Logger = loggerFactory.CreateLogger<Program>();
+        }
+
+        public ILogger Logger
+        {
+            get { return _logger; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+
+                _logger = value;
+            }
+        }
+
+        public CommandOutputProvider CommandOutputProvider
+        {
+            get { return _loggerProvider; }
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+
+                _loggerProvider = value;
+            }
+        }
+
+        public int Main(string[] args)
+        {
+            try
+            {
+                var app = new CommandLineApplication();
+                app.Description = "Manages user secrets";
+
+                app.HelpOption("-?|-h|--help");
+                var optVerbose = app.Option("-v|--verbose", "Verbose output", CommandOptionType.NoValue);
+
+                app.Command("set", c =>
+                {
+                    var optionProject = c.Option("-p|--project <PATH>", "Path to the project", CommandOptionType.SingleValue);
+                    var keyArg = c.Argument("<name>", "Name of the key");
```

Comment:
I would suggest we follow the convention used by other command-line tools like `dnu`, use `[name]` to represent argument value name